### ISSUE TITLE
Rename partition column name succeeds, but does nothing

### ISSUE THREAD

**Issue by nezihyigitbasi** [prestodb/presto#6063]
Renaming a partition column of a Hive table returns successfully but is a noop. It would be nice to return a meaningful message to the user. In this case Hive fails with `FAILED: Execution Error, return code 1 from org.apache.hadoop.hive.ql.exec.DDLTask. Invalid column reference <col_name>`.

```
presto:nyigitbasi> desc presto_insert_test;
 Column |         Type         |    Comment
--------+----------------------+---------------
 x      | bigint               |
 arr    | array(bigint)        |
 map    | map(varchar, bigint) |
 y      | varchar              | Partition Key
(4 rows)

presto:nyigitbasi> alter table presto_insert_test rename column y to new_y;
RENAME COLUMN

presto:nyigitbasi> desc presto_insert_test;
 Column |         Type         |    Comment
--------+----------------------+---------------
 x      | bigint               |
 arr    | array(bigint)        |
 map    | map(varchar, bigint) |
 y      | varchar              | Partition Key
(4 rows)

Query 20160907_185625_00003_n39u7, FINISHED, 1 node
http://localhost:8080/query.html?20160907_185625_00003_n39u7
Splits: 2 total, 2 done (100.00%)
CPU Time: 0.0s total,    2K rows/s,  154KB/s, 100% active
Per Node: 0.0 parallelism,    17 rows/s, 1.33KB/s
Parallelism: 0.0
0:00 [4 rows, 315B] [17 rows/s, 1.33KB/s]
```

**Comment by electrum** [comment]
Are we calling the metastore? If so, is it just ignoring the request?

**Comment by nezihyigitbasi** [comment]
Stepping through the code I think the problem is [here](https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/metastore/BridgingHiveMetastore.java#L141-L146). This for loop only looks at the regular columns (`table.getSd().getCols()`), not the partition columns (`table.getPartitionKeys()`). 

I can send a PR after we decide on how to handle partition keys.

**Comment by electrum** [comment]
If the metastore allows it and it works, we should fix it, otherwise we can throw a "not supported" error.

**Comment by nezihyigitbasi** [comment]
Metastore doesn't allow it:
```
Caused by: org.apache.hadoop.hive.metastore.api.InvalidOperationException: partition keys can not be changed.
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$alter_table_result$alter_table_resultStandardScheme.read(ThriftHiveMetastore.java:38648)
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$alter_table_result$alter_table_resultStandardScheme.read(ThriftHiveMetastore.java:38634)
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$alter_table_result.read(ThriftHiveMetastore.java:38576)
        at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:78)
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.recv_alter_table(ThriftHiveMetastore.java:1316)
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.alter_table(ThriftHiveMetastore.java:1301)
        at com.facebook.presto.hive.ThriftHiveMetastoreClient.alterTable(ThriftHiveMetastoreClient.java:105)
        at com.facebook.presto.hive.metastore.ThriftHiveMetastore.lambda$alterTable$9(ThriftHiveMetastore.java:304)
        at com.facebook.presto.hive.metastore.HiveMetastoreApiStats.lambda$wrap$0(HiveMetastoreApiStats.java:42)
        at com.facebook.presto.hive.RetryDriver.run(RetryDriver.java:137)
        at com.facebook.presto.hive.metastore.ThriftHiveMetastore.alterTable(ThriftHiveMetastore.java:298)
        ... 13 more
```

**Comment by haozhun** [comment]
@electrum Why should this be "not supported"? I thought not supported means a feature we may implement in the future.

**Comment by electrum** [comment]
It's also used for operations that are fundamentally not supported.

**Comment by haozhun** [comment]
Merged #6073.
